source/common/router/router.cc

```diff
@@ -2129,10 +2129,14 @@ void Filter::maybeProcessOrcaLoadReport(const Envoy::Http::HeaderMap& headers_or
   }
   // Check whether we need to send the load report to the LRS or invoke the ORCA
   // callbacks.
-  auto host = upstream_request.upstreamHost();
-  const bool need_to_send_load_report =
-      (host != nullptr) && cluster_->lrsReportMetricNames().has_value();
-  if (!need_to_send_load_report && orca_load_report_callbacks_.expired()) {
+  OptRef<const Upstream::HostDescription> upstream_host = upstream_request.upstreamHost();
+  if (!upstream_host.has_value()) {
+    return;
```

Comment:
I think this can be tricky to debug later. I suggest at least adding a debug line in this case.

Reply:
Maybe a ENVOY_BUG, may shouldn't happen in normal case.